src/multisig.ts

```diff
@@ -248,8 +248,8 @@ export function mergeMultisigTransactions(multisigTxnBlobs: Uint8Array[]) {
     address.fromMultisigPreImg(refPreImage)
   );
 
-  let newSubsigs = refSigTx.msig.subsig;
-  for (let i = 0; i < multisigTxnBlobs.length; i++) {
+  const newSubsigs = refSigTx.msig.subsig;
```

Comment:
Since this new code modifies `newSubsigs` directly, I'd feel better if we took a deep copy of `refSigTx.msig.subsig` instead of just an alias. This should do it:
```suggestion
  const newSubsigs: EncodedSubsig[] = refSigTx.msig.subsig.map(({ pk, s }) => ({
    pk,
    s,
  }));
```

Reply:
Do you feel strongly about this? map will do a shallow copy of the array, and destructuring pk and s and returning them in a new object won't create new Uint8Arrays for their values, they will still point to the same reference. For a deep copy I'd need to instantiate new Uint8Arrays , add a check to see if s is undefined etc. It adds clutter + takes some extra memory.

Reply:
Since we never modify the Uint8Arrays I'm ok with them not being copied.

I think this suggestion, while not strictly necessary, will reduce the likelihood of future mistakes, since it's not obvious that `refSigTx.msig.subsig` is being muted along with `newSubsigs`. Alternatively, if you wanted to add a comment above the definition of `newSubsigs` that modifications to it also affect `refSigTx.msig.subsig`, I view that as acceptable also.

Reply:
I kept the spirit of your snippet but didn't commit it as is. By destructuring pk and s and reinjecting them in a new object there was a situation when s didn't exist in the original obj, it would be set as an undefined prop in the new obj, and then to null probably during msgpack encoding at the bottom at the func. This made a bunch of unit-test fails since it wasn't expected. One of those things... 😅

Reply:
Good catch, thanks for making that change